src/fsharp/FSharp.Core/array.fs

```diff
@@ -405,10 +405,12 @@ namespace Microsoft.FSharp.Collections
             loop 0
 
         let inline groupByImpl (comparer:IEqualityComparer<'SafeKey>) (keyf:'T->'SafeKey) (getKey:'SafeKey->'Key) (array: 'T[]) =
+            let length = array.Length
+            if length = 0 then Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked 0 else
```

Comment:
Maybe it's time to add empty arrays singleton? https://github.com/Microsoft/visualfsharp/pull/5066#issuecomment-394110037

Reply:
yes. I could use them in many places